File: app/src/test/java/com/lucas/omnia/activities/SearchLawActivityTest.java
package com.lucas.omnia.activities;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SearchLawActivityTest {
    private final SearchLawActivity searchLawActivity = new SearchLawActivity();

    @Test
    public void searchLawActivity_ValidQuerySimple_ReturnsUrl() {
        assertEquals(searchLawActivity.composeQuery("data"), "https://www.lexml.gov" +
                ".br/busca/SRU?query=urn=lei&query=description=\"data\"");
    }

    @Test
    public void searchLawActivity_ValidQueryComposed_ReturnsUrl() {
        assertEquals(searchLawActivity.composeQuery("data law"), "https://www.lexml.gov" +
                ".br/busca/SRU?query=urn=lei&query=description=\"data+law\"");
    }
}

File: app/src/test/java/com/lucas/omnia/activities/SearchLawActivityTest.kt
package com.lucas.omnia.activities

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class SearchLawActivityTest {
    private val searchLawActivity = SearchLawActivity()
    @Test
    fun searchLawActivity_ValidQuerySimple_ReturnsUrl() {
        Assertions.assertEquals(
            searchLawActivity.composeQuery("data"), "https://www.lexml.gov" +
                    ".br/busca/SRU?query=urn=lei&query=description=\"data\""
        )
    }

    @Test
    fun searchLawActivity_ValidQueryComposed_ReturnsUrl() {
        Assertions.assertEquals(
            searchLawActivity.composeQuery("data law"), "https://www.lexml.gov" +
                    ".br/busca/SRU?query=urn=lei&query=description=\"data+law\""
        )
    }
}